<a href="https://colab.research.google.com/github/MegaSapiens/uAgentsMS/blob/main/SA_AI_bot.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

https://t.me/SA_AI_bot

In [2]:
from google.colab import userdata
fetAPI = userdata.get('fet_api_key')
botToken = userdata.get('bot_token')
#fetAPI, botToken

In [ ]:
!pip install pyTelegramBotAPI

In [ ]:
#!/usr/bin/env python
import telebot
def get_message(message):
  print(message)
  return message.content_type # + input(">")

bot = telebot.TeleBot(botToken)
c_types=['text', 'audio', 'document', 'photo', 'sticker', 'video', 'video_note', 'voice', 'location', 'contact', 'pool', 'thumbnail']
@bot.message_handler(content_types=c_types)
def echo(message):
  bot.send_message(message.chat.id, get_message(message))

In [ ]:
bot.polling()

{'content_type': 'text', 'id': 334, 'message_id': 334, 'from_user': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'username': 'SRLBP', 'last_name': 'СиРеЛюБиПро', 'language_code': 'ru', 'can_join_groups': None, 'can_read_all_group_messages': None, 'supports_inline_queries': None, 'is_premium': None, 'added_to_attachment_menu': None}, 'date': 1702146272, 'chat': {'id': 263215675, 'type': 'private', 'title': None, 'username': 'SRLBP', 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'is_forum': None, 'photo': None, 'bio': None, 'join_to_send_messages': None, 'join_by_request': None, 'has_private_forwards': None, 'has_restricted_voice_and_video_messages': None, 'description': None, 'invite_link': None, 'pinned_message': None, 'permissions': None, 'slow_mode_delay': None, 'message_auto_delete_time': None, 'has_protected_content': None, 'sticker_set_name': None, 'can_set_sticker_set': None, 'linked_chat_id': None, 'location': None, 'active_usernames': None, 'emoji_status_custo

In [ ]:
#!/usr/bin/env python
import requests # Импорт библиотеки requests для отправки HTTP-запросов
from time import sleep  # Импорт функции sleep для приостановки выполнения программы на несколько секунд

# Класс для взаимодействия с Телеграм ботом
class ForTelega():
  #
  def __init__(self, botToken):
    self.BOT_API_URL = f"https://api.telegram.org/bot{botToken}"
    response = requests.get(f"{self.BOT_API_URL}/getMe")
    if response.status_code == 200:
      print("GetMe Ok!", response.json()['result'])
    else:
      print("Error:", response.status_code)
    self.respGetMe = response
    self.offset = 0

  #
  def getUpdate(self):
    if len(self.updates) == 0:
      self.updates = requests.get(f"{BOT_API_URL}/getUpdates?offset={self.offset}").json()["result"]
      print(updates)
      if updates.status_code == 200:
        # обработка обновлений
        if len(updates["result"]) > 0:
          last_update = updates["result"][-1]
          self.offset = last_update["update_id"] + 1
          self.updates = updates["result"]
          print(updates)
        else:
          print ["Error:", updates.status_code]



    updates_url = f"{self.BOT_API_URL}/getUpdates?offset={self.offset}"
    response = requests.get(updates_url)
    if response.status_code == 200:
      updates = response.json()["result"] # обновления получены с помощью getUpdates
      return updates
    else:
       return ["Error:", response.status_code]

  #
  def get_message(self):
    for update in self.getUpdate():
      if 'message' in update and 'text' in update['message']:
        text = update['message']['text']
        if text == '/start':   # Обработка команды /start
          chat_id = update['message']['chat']['id']
          response = requests.post(f"{BOT_API_URL}/sendMessage", json={"chat_id": chat_id,
              "text": "Привет! Я бот, который умеет отвечать на вопросы. Задавайте вопросы на русском языке."})
        else: # Во всех других случаях
          response = requests.post(f"{BOT_API_URL}/sendMessage", json={"chat_id": chat_id,
              "text": f"Ok! {text}"})

In [3]:
#!/usr/bin/env python
import requests # Импорт библиотеки requests для отправки HTTP-запросов
from time import sleep  # Импорт функции sleep для приостановки выполнения программы на несколько секунд

BOT_API_URL = f"https://api.telegram.org/bot{botToken}"

response = requests.get(f"{BOT_API_URL}/getMe")

if response.status_code == 200:
  print("GetMe Ok!", response.json()['result'])
else:
  print("Error:", response.status_code)

GetMe Ok! {'id': 6332506804, 'is_bot': True, 'first_name': 'Super Agent AI', 'username': 'SA_AI_bot', 'can_join_groups': True, 'can_read_all_group_messages': False, 'supports_inline_queries': True}


In [ ]:
response.cookies

<RequestsCookieJar[]>

In [ ]:
updates_url = f"{BOT_API_URL}/getUpdates"

response = requests.get(updates_url)

if response.status_code == 200:
  updates = response.json()["result"] # получить обновления с помощью getUpdates
  print(updates)
else:
  print("Error:", response.status_code)

[{'update_id': 954598644, 'message': {'message_id': 354, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702150675, 'animation': {'file_name': 'Fuuu2.mp4', 'mime_type': 'video/mp4', 'duration': 1, 'width': 1280, 'height': 1280, 'thumbnail': {'file_id': 'AAMCAgADGQEAAgFiZXTCEyBISZC1sfGjLGN6zmoqvKYAAtwQAALpYUlIj5Bfokd_38wBAAdtAAMzBA', 'file_unique_id': 'AQAD3BAAAulhSUhy', 'file_size': 11028, 'width': 320, 'height': 320}, 'thumb': {'file_id': 'AAMCAgADGQEAAgFiZXTCEyBISZC1sfGjLGN6zmoqvKYAAtwQAALpYUlIj5Bfokd_38wBAAdtAAMzBA', 'file_unique_id': 'AQAD3BAAAulhSUhy', 'file_size': 11028, 'width': 320, 'height': 320}, 'file_id': 'CgACAgIAAxkBAAIBYmV0whMgSEmQtbHxoyxjes5qKrymAALcEAAC6WFJSI-QX6JHf9_MMwQ', 'file_unique_id': 'AgAD3BAAAulhSUg', 'file_size': 291031}, 'document': {'file_name

In [ ]:
for update in updates:
  print(update)
  if 'message' in update and 'text' in update['message']:
    text = update['message']['text']
    if text == '/start':
      chat_id = update['message']['chat']['id']  # Обработка команды /start
      response = requests.post(f"{BOT_API_URL}/sendMessage", json={"chat_id": chat_id,
          "text": "Привет! Я бот, который умеет отвечать на вопросы. Задавайте вопросы на русском языке."})
    else:
      response = requests.post(f"{BOT_API_URL}/sendMessage", json={"chat_id": chat_id,
          "text": f"Ok! {text}"})

{'update_id': 954598644, 'message': {'message_id': 354, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702150675, 'animation': {'file_name': 'Fuuu2.mp4', 'mime_type': 'video/mp4', 'duration': 1, 'width': 1280, 'height': 1280, 'thumbnail': {'file_id': 'AAMCAgADGQEAAgFiZXTCEyBISZC1sfGjLGN6zmoqvKYAAtwQAALpYUlIj5Bfokd_38wBAAdtAAMzBA', 'file_unique_id': 'AQAD3BAAAulhSUhy', 'file_size': 11028, 'width': 320, 'height': 320}, 'thumb': {'file_id': 'AAMCAgADGQEAAgFiZXTCEyBISZC1sfGjLGN6zmoqvKYAAtwQAALpYUlIj5Bfokd_38wBAAdtAAMzBA', 'file_unique_id': 'AQAD3BAAAulhSUhy', 'file_size': 11028, 'width': 320, 'height': 320}, 'file_id': 'CgACAgIAAxkBAAIBYmV0whMgSEmQtbHxoyxjes5qKrymAALcEAAC6WFJSI-QX6JHf9_MMwQ', 'file_unique_id': 'AgAD3BAAAulhSUg', 'file_size': 291031}, 'document': {'file_name'

In [7]:
import time

offset = 0

while True:
  updates = requests.get(f"{BOT_API_URL}/getUpdates?offset={offset}").json()


  print(offset, updates)
  #input(f">")

  time.sleep(5) # задержка в 5 секунду

  # обработка обновлений

  if len(updates["result"]) > 0:
    last_update = updates["result"][-1]
    offset = last_update["update_id"] + 1

0 {'ok': True, 'result': [{'update_id': 954598647, 'message': {'message_id': 357, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702200346, 'text': 'f'}}]}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': []}
954598648 {'ok': True, 'result': [{'update_id': 954598648, 'edited_message': {'message_id': 353, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702150505, 'edit_date': 1702200420, 'for

KeyboardInterrupt: ignored

`[{'update_id': 954598563, 'message': {'message_id': 204, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702137789, 'text': '/start', 'entities': [{'offset': 0, 'length': 6, 'type': 'bot_command'}]}}, {'update_id': 954598564, 'message': {'message_id': 205, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702138198, 'text': 'ввввв'}}, {'update_id': 954598565, 'message': {'message_id': 206, 'from': {'id': 263215675, 'is_bot': False, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'language_code': 'ru'}, 'chat': {'id': 263215675, 'first_name': 'СРЛБП', 'last_name': 'СиРеЛюБиПро', 'username': 'SRLBP', 'type': 'private'}, 'date': 1702138297, 'text': 'тест1'}}]`

In [ ]:
last_update_id = response["result"][-1]["update_id"]

if response["result"][-1]["message"]["text"] == "/start":
  send_message(chat_id, "Привет! Я бот!")

Программа для общения с моделью искусственного интеллекта

In [ ]:
import requests  # Импорт библиотеки requests для отправки HTTP-запросов
from time import sleep  # Импорт функции sleep для приостановки выполнения программы на несколько секунд


URL = "https://ai.fakeopen.com/v1/chat/completions"  # URL для отправки запросов к API
HEADERS = {  # Заголовки для запроса
    "Content-Type": 'application/json',
    "Authorization": 'Bearer pk-this-is-a-real-free-pool-token-for-everyone',
    "Origin": 'https://chat.geekgpt.org',
    "Referrer": 'https://chat.geekgpt.org/',
}

# Настройки для запроса к API
SETTINGS = {
    "frequency_penalty": 0,
    "messages": [],  # Список сообщений, который будет заполняться в процессе выполнения программы
    "model": "gpt-3.5-turbo",
    "presence_penalty": 0,
    "stream": False,
    "temperature": 1,
    "top_p": 1,
}

MAX_RETRIES = 50  # Максимальное количество попыток повторной отправки запроса

messages = [  # Начальное системное сообщение
    {
        "content": "Ты большая языковая модель - специалист в области IT (Python, Go, DevOpps, и т.п.), Внимательно слушай пользователя и отвечай на русском языке.",
        "role": "system",
    }
]


def send_message(msg):
    # Функция для отправки сообщения через API
    data = SETTINGS.copy()  # Создание копии настроек
    data['messages'] = messages.copy()  # Копирование списка сообщений
    data['messages'].append(msg)  # Добавление нового сообщения в список
    return requests.post(URL, json=data, headers=HEADERS)  # Отправка POST-запроса


if __name__ == '__main__':
    while True:
        new_content = input()  # Ожидание ввода нового сообщения от пользователя
        message = {  # Создание объекта сообщения
            "content": new_content,
            "role": "user",
        }
        resp = send_message(message)  # Отправка сообщения через функцию send_message
        data = resp.json()  # Получение данных из ответа в формате JSON
        answers = data.get("choices")  # Получение вариантов ответов из данных
        retries = 1
        while not answers:  # Цикл повторной отправки запроса в случае отсутствия ответов
            retries += 1
            if retries > MAX_RETRIES:  # Если количество попыток превышает максимальное
                break
            print(">", end="", flush=True)  # Запрос с ошибкой, ждем 3 сек")  # Вывод сообщения о повторной отправке
            sleep(3)  # Приостановка выполнения программы на 5 секунд
            resp = send_message(message)  # Повторная отправка сообщения
            data = resp.json()  # Получение данных из ответа
            answers = data.get("choices")  # Получение вариантов ответов из повторного ответа

        if answers:  # Если есть варианты ответов
            answer = answers[0]['message']  # Выбор первого варианта ответа
            #print(f"Токенов: {data['usage']}")  # Вывод информации о количестве использованных токенов
            print(f"Ответ: {answer['content']}")  # Вывод выбранного ответа
            messages.append(message)  # Добавление сообщения пользователя в список сообщений
            messages.append(answer)  # Добавление ответа в список сообщений
        else:  # Если ответ не содержит вариантов
            print(f"Ошибка в запросе: {data}")  # Вывод сообщения об ошибке


>>>>>>>>>>>Ответ: Нет одного "идеально-универсального" языка программирования, который подходит для всех задач. Однако существуют языки программирования, которые обладают определенными особенностями и могут использоваться в различных сферах:

1. **Python:** Известен своей простотой и читаемостью кода, что делает его отличным выбором для начинающих. Он широко используется в веб-разработке, науке о данных, искусственном интеллекте, автоматизации и многих других областях.

2. **JavaScript:** Основной язык для веб-разработки. Он позволяет создавать интерактивные веб-страницы и веб-приложения. Также он используется для разработки серверных приложений (Node.js).

3. **Java:** Популярен в корпоративной среде. Используется для создания мобильных приложений (Android), больших систем, веб-приложений и другого программного обеспечения.

4. **C/C++:** Имеют высокую производительность и используются в системном программировании, разработке операционных систем, играх, встраиваемых системах и других 